File: api-cmssignature/api-mssclient/mssclient/csharp/src/wrapper/signer/BinarySignable.cs
using System;

namespace tr.gov.tubitak.uekae.esya.api.webservice.mssclient.wrapper.signer
{
    /**
 *  Represents signable base64 encoded binary data
 */

    public class BinarySignable : ISignable
    {
        private readonly String _base64EncodedHashVal;
        private readonly String _dataToBeDisplayed;

        private readonly SignatureType _signatureType = SignatureType.PKCS7;
        private readonly String mimeType = "application/octet-stream";
        private readonly String encoding = "Base64";
        private String hashURI;

        public BinarySignable(String aBase64EncodedHashVal, String aDataToBeDisplayed, SignatureType aSignatureType)
        {
            _base64EncodedHashVal = aBase64EncodedHashVal;
            _dataToBeDisplayed = aDataToBeDisplayed;
            if (aSignatureType != null)
                _signatureType = aSignatureType;
        }

        public BinarySignable(String aBase64EncodedHashVal, String aDataToBeDisplayed, SignatureType aSignatureType, String aMimeType, String aEncoding)
        {
            _base64EncodedHashVal = aBase64EncodedHashVal;
            _dataToBeDisplayed = aDataToBeDisplayed;
            _signatureType = aSignatureType;
            mimeType = aMimeType;
            encoding = aEncoding;

        }

        public String getValueToBeDisplayed()
        {
            return _dataToBeDisplayed;
        }

        public String getValueToBeSigned()
        {
            return _base64EncodedHashVal;
        }

        public String getEncoding()
        {
            return encoding;
        }

        public String getMimeType()
        {
            return mimeType;
        }

        public SignatureType getSignatureType()
        {
            return _signatureType;
        }

        public String getHashURI()
        {
            return hashURI;
        }

        public void setHashURI(String hashURI)
        {
            this.hashURI = hashURI;
        }
    }
}

File: api-cmssignature/api-mssclient/mssclient/src/tr/gov/tubitak/uekae/esya/api/webservice/mssclient/wrapper/signer/BinarySignable.java
package tr.gov.tubitak.uekae.esya.api.webservice.mssclient.wrapper.signer;

import tr.gov.tubitak.uekae.esya.api.webservice.mssclient.wrapper.SignatureType;

/**
 *  Represents signable base64 encoded binary data
 */
public class BinarySignable implements ISignable {
    private String base64EncodedHashVal;
    private String dataToBeDisplayed;
    private SignatureType signatureType = SignatureType.PKCS7;
    private String hashURI;
    private String mimeType = "application/octet-stream";
    private String encoding = "Base64"; 


    public BinarySignable(String aBase64EncodedHashVal, String aDataToBeDisplayed, SignatureType aSignatureType) {
        base64EncodedHashVal = aBase64EncodedHashVal;
        dataToBeDisplayed = aDataToBeDisplayed; 
        signatureType = aSignatureType;
    }
    
    public BinarySignable(String aBase64EncodedHashVal, String aDataToBeDisplayed, SignatureType aSignatureType, String aMimeType, String aEncoding) {
        base64EncodedHashVal = aBase64EncodedHashVal;
        dataToBeDisplayed = aDataToBeDisplayed;        
        signatureType = aSignatureType;
        mimeType = aMimeType;
        encoding = aEncoding;
        
    }

    public String getValueToBeDisplayed() {
        return dataToBeDisplayed;
    }

    public String getValueToBeSigned() {
        return base64EncodedHashVal;
    }

    public String getEncoding() {
        return encoding;
    }

    public String getMimeType() {
        return mimeType;
    }

    public SignatureType getSignatureType() {
        return signatureType;
    }

    public String getHashURI(){
        return hashURI;
    }

    public void setHashURI(String hashURI){
        this.hashURI = hashURI;
    }
}
